src/integration_tests/obstacle_avoidance_control.cpp

```
@@ -0,0 +1,149 @@
+                std::this_thread::sleep_for(std::chrono::seconds(1));
+                run_cmd("python3 src/integration_tests/test_data/obstacle_avoidance_dummy.py &");
+                break;
+            default:
+                LogInfo() << "Unsupported control type";
+        }
+    });
+
+    // Start controls sending
+    {
+        LogInfo() << "Starting obstacle avoidance...";
+        std::promise<void> prom1;
+        std::future<void> fut1 = prom1.get_future();
+        obstacle_avoidance->start_async([&prom1](ObstacleAvoidance::Result result) {
+            EXPECT_EQ(result, ObstacleAvoidance::Result::Success);
+            prom1.set_value();
+        });
+        EXPECT_EQ(fut1.wait_for(std::chrono::seconds(2)), std::future_status::ready);
+    }
+
+    std::this_thread::sleep_for(std::chrono::seconds(3));
+
+    {
+        LogInfo() << "Restarting obstacle avoidance...";
+        std::promise<void> prom;
+        std::future<void> fut = prom.get_future();
+        obstacle_avoidance->restart_async([&prom](ObstacleAvoidance::Result result) {
+            EXPECT_EQ(result, ObstacleAvoidance::Result::Success);
+            prom.set_value();
+        });
+        EXPECT_EQ(fut.wait_for(std::chrono::seconds(2)), std::future_status::ready);
+    }
+
+    std::this_thread::sleep_for(std::chrono::seconds(5));
+
+    {
+        LogInfo() << "Stopping obstacle avoidance...";
+        std::promise<void> prom;
+        std::future<void> fut = prom.get_future();
+        obstacle_avoidance->stop_async([&prom](ObstacleAvoidance::Result result) {
+            EXPECT_EQ(result, ObstacleAvoidance::Result::Success);
+            prom.set_value();
+        });
+        EXPECT_EQ(fut.wait_for(std::chrono::seconds(3)), std::future_status::ready);
+    }
+}
+
+void run_cmd(const std::string& cmd_str)
+{
+    const char* cmd = cmd_str.c_str();
+    if (system(NULL)) {
+        system(cmd);
+    }
```

What's this `system(NULL)` check for? And if it fails, then this will silently fail and you'll have a hard time debugging it.